### ISSUE TITLE
[Bug]: Milvus crashed when loading with json field and growing segments

### ISSUE THREAD

**Issue by binbinlv** [milvus-io/milvus#24566]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.2.9 latest
- Deployment mode(standalone or cluster): both
- MQ type(rocksmq, pulsar or kafka):    all
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus2.2.9.dev30
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Milvus crashed when loading with json field and growing segments
panic
```
[2023/05/31 08:02:17.199 +00:00] [INFO] [querynode/segment_loader.go:220] ["start to load segments in parallel"] [collectionID=441848980713442913] [segmentType=Growing] [segmentNum=1] [concurrencyLevel=1]
[2023/05/31 08:02:17.199 +00:00] [INFO] [querynode/segment_loader.go:260] ["start loading segment data into memory"] [collectionID=441848980713442913] [partitionID=441848980713442914] [segmentID=441848980713442922] [segmentType=Growing]
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0x1010001fe pc=0x7f8e53b246f7]

runtime stack:
runtime.throw({0x4073407?, 0x7f8b7f280440?})
	/usr/local/go/src/runtime/panic.go:992 +0x71
runtime.sigpanic()
	/usr/local/go/src/runtime/signal_unix.go:802 +0x3a9

goroutine 6428 [syscall]:
runtime.cgocall(0x3143050, 0xc00180f5f8)
	/usr/local/go/src/runtime/cgocall.go:157 +0x5c fp=0xc00180f5d0 sp=0xc00180f598 pc=0x147e57c
github.com/milvus-io/milvus/internal/querynode._Cfunc_LoadFieldData(0x7f8c226a7000, 0x7f8c226092a0)
	_cgo_gotypes.go:692 +0x55 fp=0xc00180f5f8 sp=0xc00180f5d0 pc=0x2dd09d5
github.com/milvus-io/milvus/internal/querynode.(*Segment).LoadMultiFieldData.func1(0x1?, 0x4063b16?)
	/go/src/github.com/milvus-io/milvus/internal/querynode/segment.go:867 +0x65 fp=0xc00180f638 sp=0xc00180f5f8 pc=0x2df0765
github.com/milvus-io/milvus/internal/querynode.(*Segment).LoadMultiFieldData(0xc002176540, 0xbb8, {0xc0049345c0, 0x7, 0x400?})
	/go/src/github.com/milvus-io/milvus/internal/querynode/segment.go:867 +0x2f5 fp=0xc00180f7a8 sp=0xc00180f638 pc=0x2df0395
github.com/milvus-io/milvus/internal/querynode.(*segmentLoader).loadFiles(0xc0023f2900, {0x44b7f38, 0xc004f20750}, 0xc002176540, 0xc005509680)
	/go/src/github.com/milvus-io/milvus/internal/querynode/segment_loader.go:313 +0x4a5 fp=0xc00180fbe0 sp=0xc00180f7a8 pc=0x2d9b805
github.com/milvus-io/milvus/internal/querynode.(*segmentLoader).LoadSegment.func3(0x0?)
	/go/src/github.com/milvus-io/milvus/internal/querynode/segment_loader.go:203 +0xe5 fp=0xc00180fdc0 sp=0xc00180fbe0 pc=0x2d9ada5
github.com/milvus-io/milvus/internal/util/funcutil.ProcessFuncParallel.func3(0x0)
	/go/src/github.com/milvus-io/milvus/internal/util/funcutil/parallel.go:83 +0x16d fp=0xc00180ffc8 sp=0xc00180fdc0 pc=0x20a908d
github.com/milvus-io/milvus/internal/util/funcutil.ProcessFuncParallel.func4()
	/go/src/github.com/milvus-io/milvus/internal/util/funcutil/parallel.go:101 +0x2a fp=0xc00180ffe0 sp=0xc00180ffc8 pc=0x20a8eea
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1571 +0x1 fp=0xc00180ffe8 sp=0xc00180ffe0 pc=0x14e7c81
created by github.com/milvus-io/milvus/internal/util/funcutil.ProcessFuncParallel
	/go/src/github.com/milvus-io/milvus/internal/util/funcutil/parallel.go:70 +0x1aa

goroutine 1 [chan receive]:
github.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).Run(0xc001a47e58, 0x1, {0x0, 0x0})
	/go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:351 +0xb0d
github.com/milvus-io/milvus/cmd/milvus.(*run).execute(0xc000f846c0, {0xc0001cc060?, 0x3, 0x3}, 0xc00048cf60)
	/go/src/github.com/milvus-io/milvus/cmd/milvus/run.go:112 +0x66e
github.com/milvus-io/milvus/cmd/milvus.RunMilvus({0xc0001cc060?, 0x3, 0x3})
	/go/src/github.com/milvus-io/milvus/cmd/milvus/milvus.go:60 +0x21e
main.main()
	/go/src/github.com/milvus-io/milvus/cmd/main.go:26 +0x2e

goroutine 391 [select]:
github.com/milvus-io/milvus/internal/config.(*EtcdSource).refreshConfigurationsPeriodically(0xc000455800)
	/go/src/github.com/milvus-io/milvus/internal/config/etcd_source.go:147 +0x9f
created by github.com/milvus-io/milvus/internal/config.(*EtcdSource).GetConfigurations.func1
	/go/src/github.com/milvus-io/milvus/internal/config/etcd_source.go:98 +0x5a
...
```

### Expected Behavior

Load successfully without panic

### Milvus Log

[binbin_load_crash.log](https://github.com/milvus-io/milvus/files/11611977/binbin_load_crash.log)

**Comment by yah01** [comment]
Fixed with #24567

**Comment by binbinlv** [comment]
Verified and fixed.

pymilvus:2.2.9.dev32
milvus:2.2.0-20230531-d163b771
